File: hw1/hw1/Calculator.cs
using System;

namespace hw1
{
    public enum Operation
    {
        Plus,
        Minus,
        Multiply,
        Divide,
        IncorrectOperation
    }

    public static class Calculator
    {
        public static double Calculate(int val1, Operation operation, int val2)
        {
            return operation switch
            {
                Operation.Plus => val1 + val2,
                Operation.Minus => val1 - val2,
                Operation.Multiply => val1 * val2,
                Operation.Divide => (double)val1 / (double)val2,
                Operation.IncorrectOperation => throw new ArgumentException("Incorrect operation")
            };
        }
    }
}

File: hw4/hw4/Calculator.fs
module hw4.Calculator

let calculate (val1:double) op (val2:double) =
    if op = Operation.Divide && val2 = 0. then
        raise Exceptions.DivisionByZero
    match op with
    | Plus -> val1 + val2
    | Minus -> val1 - val2
    | Multiply -> val1 * val2
    | Divide -> val1 / val2